plasmapy/diagnostics/thomson.py

```diff
@@ -0,0 +1,208 @@
+                     ion_vel=None,
+                     probe_vec=np.array([1, 0, 0]),
+                     scatter_vec=np.array([0, 1, 0])):
+    r"""
+    Calculate the spectral dispersion function for Thomson scattering of a
+    probe laser beam by a multi-species Maxwellian plasma.
+
+    Parameters
+    ---------
+
+    wavelengths : astropy.units.Quantity ndarray (required)
+        Array of wavelengths over which the spectral density function
+        will be calculated, convertable to nm.
+
+    probe_wavelength : astropy.units.Quantity (required)
+        Wavelength of the probe laser convertable to nm
+
+    ne : astropy.units.Quantity, ndarray (required)
+        mean (0th order) electron density of all plasma components combined
+        convertable to cm^-3.
+
+    kTe : astropy.units.Quantity (required)
+        Temperature of the electron component convertable to eV.
+
+    kTi : astropy.units.Quantity ndarray, shape [N] (required)
+        Temperature of each ion component convertable to eV.
+
+    fract : float ndarray, shape [N]
+        Fraction (by number) of the total number of ions made up by each ion
+        species. Must sum to 1.0. Default is a single ion species.
+
+    ion_species : str ndarray, shape [N]
+        Strings representing each ion species in the format interpretable by
+        the plasmapy Particle class. Default is ['H+'] corresponding to a
+        single species of hydrogen ions.
+
+    fluid_vel : astropy.units.Quantity ndarray shape [3]
+        Velocity of the fluid or electron component in the rest frame in
+        units convertable to cm/s. Defaults to [0, 0, 0], representing
+        a stationary plasma.
+
+    ion_vel : astropy.units.Quantity ndarray, shape [N,3]
+        Velocity vectors for each ion population relative to the
+        fluid velocit in units convertable to cm/s. Defaults zero drift
+        for all specified ion species.
+
+    probe_vec : float ndarray, shape [3]
+        Unit vector in the direction of the probe laser. Defaults to
+        [1, 0, 0].
+
+    scatter_vec : float ndarray, shape [3]
+        Unit vector pointing from the scattering volume to the detector.
+        Defaults to [0, 1, 0] which, along with the default probe_vec,
+        corresponds to a 90 degree scattering angle geometry.
+
+    Returns
+    -------
+    alpha : float
+        mean scattering parameter. alpha > 1 corresponds to collective
+        scattering, while alpha < 1 indicates non-collective scattering.
+
+    Skw : astropy.units.Quantity ndarray
+        Computed spectral density function over the input wavelength range
+        with units of nm.
+
+    Notes
+    -----
+    For details, see "Plasma Scattering of Electromagnetic Radiation" by
+    Sheffield et al. ISBN 978-0123748775. This code is a modified version of
+    the program described therein.
+
+    For a concise summary of the relevant physics, see Chapter 5 of Derek
+    Schaeffer's thesis, DOI: 10.5281/zenodo.3766933
+
```

Comment:
```suggestion
    For details, see "Plasma Scattering of Electromagnetic Radiation" by
    Sheffield et al. `ISBN 978\\-0123748775`_. This code is a modified version of
    the program described therein.

    For a concise summary of the relevant physics, see Chapter 5 of Derek
    Schaeffer's thesis, DOI: `10.5281/zenodo.3766933`_.
    
    .. _`ISBN 978\\-0123748775`: https://www.sciencedirect.com/book/9780123748775/plasma-scattering-of-electromagnetic-radiation
    .. _`10.5281/zenodo.3766933`: https://doi.org/10.5281/zenodo.3766933
```

Add links for resources.